### ISSUE TITLE
фыв

### ISSUE THREAD

**Issue by GeorgyKonoplich** [GeorgyKonoplich/BiLSTM_for_NER#1]
model = model_helper.ModelHelper(name="rnn")

        input_blob, seq_lengths, hidden_init, cell_init, target = \
            model.net.AddExternalInputs(
                'input_blob',
                'seq_lengths',
                'hidden_init',
                'cell_init',
                'target',
            )

        hidden_output_all, self.hidden_output, _, self.cell_state = LSTM(
            model, input_blob, seq_lengths, (hidden_init, cell_init),
            3, self.hidden_size, scope="LSTM")
        output = brew.fc(
            model,
            hidden_output_all,
            None,
            dim_in=self.hidden_size,
            dim_out=1,
            axis=2
        )
[char_rnn.py example](https://github.com/caffe2/caffe2/blob/master/caffe2/python/examples/char_rnn.py)
        pred = model.net.Sigmoid(output, 'pred')
        #[softmax, loss] = model.net.SoftmaxWithLoss([pred, target], ["softmax", "loss"])
        self.forward_net = core.Net(model.net.Proto())
        #print(model.net.Proto())
        #print(workspace.FetchBlob('softmax').shape)
        xent = model.net.SigmoidCrossEntropyWithLogits([softmax, target], 'xent')
        # Loss is average both across batch and through time
        # Thats why the learning rate below is multiplied by self.seq_length
        
        loss1 = model.net.AveragedLoss(loss, 'loss1')
        model.AddGradientOperators([loss1])
        
        # use build_sdg function to build an optimizer
        build_sgd(
            model,
            base_learning_rate=0.1,# * self.seq_length,
            policy="step",
            stepsize=1,
            gamma=0.9999
        )
